remove redirect for directory without trailing slash

Johann150 opened the pull request:
As discussed recently [on the mailing list](https://lists.orbitalfox.eu/archives/gemini/2020/003404.html), this is a bad practice and in this case not even necessary as `PathBuf` handles the path the same no matter if it has a trailing slash or not.

mbrubeck reviewed (changes requested):
The linked email is correct that an empty path is equivalent to the path `"/"`.  However, the path `"foo"` is not equivalent to the path `"foo/"`.

In particular, if you load the same content via the URIs `gemini://foo/bar` and `gemini://foo/bar/`, then a relative link like `=> baz` will resolve to the path `/foo/baz` in one case, and `/foo/bar/baz` in the other.

For this reason, I would prefer to keep the redirect in cases where the path is non-empty.  However, I agree it would be good to skip the redirect in the specific case of an empty path.  Would you like to change this PR to implement this?

Johann150 commented:
I have changed it, was a bit too eager there. Thanks for reviewing carefully!

I also noticed that the check for the trailing slash was not correct and fixed that. It would have checked if the complete URL ended with a slash, not if the path part ended with a slash. This would be problematic if the URL contains a query or fragment part.

mbrubeck commented:
Included in Agate 1.3.0:

https://github.com/mbrubeck/agate/releases/tag/v1.3.0

Thanks!
